src/commands.rs

```diff
@@ -545,6 +545,44 @@ implement_commands! {
     }
 
     // list operations
+    
+    /// Pop an element from the front of a list, push it to the front of another list
+    /// and return it; or block until one is available
+    fn blmove_left_to_left<K: ToRedisArgs>(srckey: K, dstkey: K, timeout: usize) {
```

Comment:
I'd collapse these four functionns into 

```
enum Direction {
    Left,
    Right,
}
fn blmove(srckey: K, dstkey: K, src_dir: Direction, dst_dir: Direction timeout: usize)
```

Reply:
Sounds good, thoughts on using this enum in the `LMPOP` and `BLMPOP` cmds and collapsing them? To keep consistency going forward when cmds use the `LEFT | RIGHT` arg.

```rs
    /// Pops `count` elements from the start of the first non-empty list key from the list of 
    /// provided key names - defaults to 1 element if count not defined
    fn lmpop_left<K: ToRedisArgs>( numkeys: usize, key: K, count: Option<core::num::NonZeroUsize>) {
        cmd("LMPOP").arg(numkeys).arg(key).arg("LEFT").arg("COUNT").arg(count)
    }

    /// Pops `count` elements from the end of the first non-empty list key from the list of 
    /// provided key names - defaults to 1 element if count not defined
    fn lmpop_right<K: ToRedisArgs>( numkeys: usize, key: K, count: Option<core::num::NonZeroUsize>) {
        cmd("LMPOP").arg(numkeys).arg(key).arg("RIGHT").arg("COUNT").arg(count)
    }
```

to 

```rs
    /// Pops `count` elements from the first non-empty list key from the list of 
    /// provided key names
    fn lmpop<K: ToRedisArgs>( numkeys: usize, key: K, dir: Direction, count: Option<core::num::NonZeroUsize>) {
        cmd("LMPOP").arg(numkeys).arg(key).arg(dir).arg("COUNT").arg(count)
    }

```

Reply:
I used the `Direction` enum in all places I could for consistency. How does it look